src/disk_interface.cc

```diff
@@ -100,19 +100,20 @@ bool StatAllFilesInDir(const string& dir, map<string, TimeStamp>* stamps,
       static_cast<FINDEX_INFO_LEVELS>(1);
   FINDEX_INFO_LEVELS level =
       can_use_basic_info ? kFindExInfoBasic : FindExInfoStandard;
-  WIN32_FIND_DATAA ffd;
```

Comment:
It's also apparently the case that "dir" is not used as a `std::string`, but just as a view into a string. Consider converting this parameter to a `StringPiece` so that callers don't need to worry about always having an actual `std::string` available.

Reply:
To clarify, I meant changing 

```
bool StatAllFilesInDir(const string& dir, map<string, TimeStamp>* stamps, string* err) {
```

to


```
bool StatAllFilesInDir(StringPiece dir, map<string, TimeStamp>* stamps, string* err) {
```

Reply:
@jlonnberg

Reply:
Which is the optimal way to concat the stringpiece with the \\* string? The change to StringPiece will require this, and it's not obvious for me on how to do it.

Reply:
std::string concat;
concat.reserve(str1.size() + str2.size() + 1);
concat.append(dir.begin(), dir.end());
concat.append(L"\\*");

Reply:
How about: 
```
 std::string dir_string;
  dir_string.reserve(dir.len_ + 3); // Make room for \* and null termination
  dir_string.append(dir.begin(),dir.end());
  dir_string.append("\\*");
```

Reply:
That's a significant improvement.

But what about this?

```
std::wstring wdir_string;
size_t wstr_size = calculate_size_of_converting_dir(dir);
wdir_string.reserve(wstr_size + 3); // Make room for \* and null termination
Utf8ToWide(dir, wdir_string); // convert directly into the wstring.
wdir_string.append(L"\\*"); // Append a wide \\*.
```